database/src/main/resources/db/migration/V1.59.0__NOTIF-512_endpoints_server_errors.sql

```diff
@@ -0,0 +1,6 @@
+-- This only affects future inserted rows but not the existing ones.
+ALTER TABLE endpoints
+    ADD COLUMN server_errors INTEGER NOT NULL DEFAULT 0;
+
+-- This updates existing rows.
+UPDATE endpoints SET server_errors = 0;
```

Comment:
shouldn't the value be already `0` as is the default and has `NOT NULL`?

Reply:
No, adding a `DEFAULT` rule to an existing table does not update old records. It only affects new records.

However, your question makes me wonder if the `NOT NULL` constraint should be added after the `UPDATE`. I'll run a quick local check for that right away.

Reply:
> No, adding a `DEFAULT` rule to an existing table does not update old records. It only affects new records.

This may be wrong. Running the `ALTER TABLE` query locally affects old records... It's not supposed to do that according to the Postgres doc but I guess the `NOT NULL` constraint changes things.

Reply:
I'll remove the `UPDATE` query from the script and we'll see how this goes on stage.

Reply:
Done.

Reply:
> but I guess the `NOT NULL` constraint changes things

That's what I recall - found this https://stackoverflow.com/a/42150852/688379